import1/dbus.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package import1 provides integration with the systemd-importd API.  See https://www.freedesktop.org/wiki/Software/systemd/importd/
+// Note: Requires systemd v231 or higher
+package import1
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/godbus/dbus"
+)
+
+const (
+	dbusInterface = "org.freedesktop.import1.Manager"
+	dbusPath      = "/org/freedesktop/import1"
+)
+
+// Conn is a connection to systemds dbus endpoint.
+type Conn struct {
+	conn   *dbus.Conn
+	object dbus.BusObject
+}
+
+// Transfer is the status for an import, export or download operations.
+type Transfer struct {
+	Id       uint32  // The numeric transfer ID of the transfer object
+	Local    string  // The local container name of this transfer
+	Remote   string  // The remote source (in case of download: the URL, in case of import/export a string describing the file descriptor passed in)
+	Type     string  // The type of operation
+	Verify   string  // The selected verification setting, and is only defined for download operations
+	Progress float64 // The current progress of the transfer, as a value between 0.0 and 1.0
```

Comment:
Is there a reason to have these as side-comments instead of normal docstrings?

Reply:
Well, just learnt from `machine1`. I have no idea on this and will update them if requested.

Reply:
Ack, I'm fine with keeping those as they are now.